pkg/provider/azure_vmssflex.go

```diff
@@ -881,10 +885,14 @@ func (fs *FlexScaleSet) EnsureBackendPoolDeletedFromVMSets(vmssNamesMap map[stri
 				},
 			}
 
-			klog.V(2).Infof("fs.EnsureBackendPoolDeletedFromVMSets begins to update vmss(%s) with backendPoolID %s", vmssName, backendPoolID)
+			defer func() {
+				fs.vmssFlexCache.Get(consts.VmssFlexKey, azcache.CacheReadTypeForceRefresh)
```

Comment:
Do we need to refresh the cache if failed to update the VMSS?

Reply:
Yes, I discovered refreshing the cache when VMSS update fails can help the operation self-healing

Reply:
I think it means we will have an extra API call when the failure is not retriable as well.

Reply:
That's true, but this is the best of what we can do. If we don't add this API call, then the VMSS updating will keep failing and this will introduce much more wasted API call. I have saw this happenning in the last few days' E2E testing.

Like Pengfei mentioned before, let's first focus on the functional correctness as top priority. After all the E2E tests pass, we can then further make optimization for the performance

Reply:
Can we only refresh the cache after a successful operation?

Reply:
No, I have tried only refreshing the cache after succussful operation, but it does not always work. The special case I discoverd is: if the cache does not reflect the latest status, then the VMSS update will keep failing again and again. 

For example, the cache says backend pool 2 is still with VMSS Flex, but actually backendpool 2 is already removed. At this time, if we want to add backend pool 5 to the VMSS, then the update will fail because it operation will request update VMSS with backend pool 2 and 5. However, since backendpool 2 is already deleted, the update operation will fail and retry will not resolve this issue until the cache expires. But before cache expires, we can see a lot of failed retry in the CCM log. This is why we need to refresh the cache even if the operation fails.

Reply:
Can we invalidate the cache by removing the entry? This could save some API calls.

Reply:
The VMSS Flex Cache stores all VMSS Flex in the cluster. There is only one entry key for this cache: https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/pkg/consts/consts.go#L149

All the VMSS Flex are queried with one batch API call using VMSS List API, similar to what we are doing for the VMSS Uniform cache

Reply:
Sure, I just modified the code to remove the entry as you suggested